channels.nix

```diff
@@ -1,87 +1,114 @@
+    };
+    "nixos-21.11-small" = {
+      job = "nixos/release-21.11-small/tested";
+      variant = "small";
+      stability = "beta";
+    };
+    "nixpkgs-21.11-darwin" = {
+      job = "nixpkgs/nixpkgs-21.11-darwin/darwin-tested";
+      variant = "darwin";
+      stability = "beta";
+    };
+    "nixos-21.11-aarch64" = {
+      job = "nixos/release-21.11-aarch64/tested";
+      variant = "aarch64";
+      stability = "beta";
+    };
 
-      "nixos-20.09" = {
-        job = "nixos/release-20.09/tested";
-        current = false;
-      };
-      "nixos-20.09-small" = {
-        job = "nixos/release-20.09-small/tested";
-        current = false;
-      };
-      "nixpkgs-20.09-darwin" = {
-        job = "nixpkgs/nixpkgs-20.09-darwin/darwin-tested";
-        current = false;
-      };
-      "nixos-20.09-aarch64" = {
-        job = "nixos/release-20.09-aarch64/tested";
-        current = false;
-      };
+    "nixos-21.05" = {
+      job = "nixos/release-21.05/tested";
+      variant = "primary";
+      stability = "stable";
+    };
+    "nixos-21.05-small" = {
+      job = "nixos/release-21.05-small/tested";
+      variant = "small";
+      stability = "stable";
+    };
+    "nixpkgs-21.05-darwin" = {
+      job = "nixpkgs/nixpkgs-21.05-darwin/darwin-tested";
+      variant = "darwin";
+      stability = "stable";
+    };
+    "nixos-21.05-aarch64" = {
+      job = "nixos/release-21.05-aarch64/tested";
+      variant = "aarch64";
+      stability = "stable";
+    };
 
-      "nixos-20.03" = {
-        job = "nixos/release-20.03/tested";
-        current = false;
-      };
-      "nixos-20.03-small" = {
-        job = "nixos/release-20.03-small/tested";
-        current = false;
-      };
-      "nixpkgs-20.03-darwin" = {
-        job = "nixpkgs/nixpkgs-20.03-darwin/darwin-tested";
-        current = false;
-      };
+    "nixos-20.09" = {
+      job = "nixos/release-20.09/tested";
+      variant = "primary";
+      stability = "unmaintained";
+    };
+    "nixos-20.09-small" = {
+      job = "nixos/release-20.09-small/tested";
+      variant = "small";
+      stability = "unmaintained";
+    };
+    "nixpkgs-20.09-darwin" = {
+      job = "nixpkgs/nixpkgs-20.09-darwin/darwin-tested";
+      variant = "darwin";
+      stability = "unmaintained";
+    };
+    "nixos-20.09-aarch64" = {
+      job = "nixos/release-20.09-aarch64/tested";
+      variant = "aarch64";
+      stability = "unmaintained";
+    };
+
+    "nixos-20.03" = {
+      job = "nixos/release-20.03/tested";
+      variant = "primary";
+      stability = "unmaintained";
+    };
+    "nixos-20.03-small" = {
+      job = "nixos/release-20.03-small/tested";
+      variant = "small";
+      stability = "unmaintained";
+    };
+    "nixpkgs-20.03-darwin" = {
+      job = "nixpkgs/nixpkgs-20.03-darwin/darwin-tested";
+      variant = "darwin";
+      stability = "unmaintained";
+    };
   };
 
   channels-with-urls = (builtins.mapAttrs (name: about: about.job) channels);
```

Comment:
Should we validate the variant and stability match the expected values here?

Reply:
I think that, since the "bounds" are described in this file, it's fine to not validate. Also, it's perfectly valid that `variant` doesn't exist (e.g. in the case of `nixpkgs-unstable`).